backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -454,6 +474,17 @@ def verify_share(
                 session, share_uri, status=status, healthStatus=healthStatus
             )
 
+        log.info(f'Verifying principal IAM Role {share.principalIAMRoleName}')
+        # If principal role doesn't exist, all share items are unhealthy, no use of further checks
+        if not ShareObjectService.verify_principal_role(session, share):
```

Comment:
IMPORTANT: In approve shares and revoke shares we are procssing share requests, we are not verifying the healthstatus (check the big comment with the picture). In approve_share, if the role is not present, the `Status` of the item should go to failure. We do not manage the unhealthyness of a share, because it has not happened yet. We could argue that it is our chance to update the customer and tell them that OTHER share items already in `Share_Succeed` are unhealthy

Reply:
Update Health message and status is our only way to show the user in UI, WHY did share fail. Otherwise it will be unclear.

Reply:
After discussions offline, we have decided to implement better UX for share failed errors in https://github.com/data-dot-all/dataall/issues/1168